### PR TITLE
Add feedback webview

### PR DESCRIPTION
**Pull request by msivasubramaniaan**
Signed-off-by: msivasubramaniaan [msivasub@redhat.com](mailto:msivasub@redhat.com)

Fix: https://github.com/redhat-developer/vscode-openshift-tools/issues/2489

### CODE REVIEW

File: src/webview/feedback/feedbackLoader.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+                sendTelemetry('feedback', event.data);
+            }
+            break;
+        default:
+            break;
+    }
+}
+
+export default class FeedbackLoader {
+    // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
```
Comment (datho7561) [inline src/webview/feedback/feedbackLoader.ts:27]: Give the function a return type instead of ignoring the linting issue

**Review by datho7561 — CHANGES_REQUESTED** [review]

**Comment by mohitsuman** [comment]
- OpenShift Logo is distorted. Need to align it properly. See screenshot.

Screenshot 2023-06-05 at 3 06 02 AM

File: src/webview/feedback/app/feedback.tsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        <div className='parentContainer'>
+            <div className='headerContainer'>
+                <header className='header__logo' style={{ width: '50%' }}>
+                    <img className='image__logo' src={OpenShiftExtensionIcon} />
+                    <div style={{ display: 'flex', flexDirection: 'column' }}>
+                        <label style={{ display: 'flex', flexDirection: 'row', textAlign: 'center' }}>
+                            <Typography variant='h4' className='highlight'>OpenShift</Typography>
+                            <Typography variant='h4' style={{ paddingLeft: '1rem' }} className='foreGroundColor'>Toolkit</Typography>
+                        </label>
+                        <Typography variant='caption' style={{ marginTop: '2px' }} className='foreGroundColor'>Your opinion matters to us!</Typography>
```
Comment (mohitsuman) [inline src/webview/feedback/app/feedback.tsx:48]: `Your opinion matters to us!` should be written beside the OpenShift logo and in a bigger font size.

File: src/webview/feedback/app/feedback.tsx [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+                                        <img src={MicrosoftIcon} />
+                                    </Icon>
+                                    <Typography variant='body2' className='footerText'>Rate us on Marketplace</Typography>
+                                </Stack>
+                            </div>
+                        </a>
+                    </label>
+                </div>
+            </div>
+            <Survey model={feedbackModal} />
```
Comment (mohitsuman) [inline src/webview/feedback/app/feedback.tsx:74]: @msivasubramaniaan The font size of the questions are very small. Let's increase the size of the Qs.

File: src/webview/feedback/app/feedback.tsx [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+                        <label style={{ display: 'flex', flexDirection: 'row', textAlign: 'center' }}>
+                            <Typography variant='h4' className='highlight'>OpenShift</Typography>
+                            <Typography variant='h4' style={{ paddingLeft: '1rem' }} className='foreGroundColor'>Toolkit</Typography>
+                        </label>
+                        <Typography variant='caption' style={{ marginTop: '2px' }} className='foreGroundColor'>Your opinion matters to us!</Typography>
+                    </div>
+                </header>
+                <div style={{ width: '50%' }}>
+                    <label style={{ display: 'flex', flexDirection: 'row', gap: '1rem', textAlign: 'center', float: 'right', margin: '1rem' }}>
+                        <a href='https://github.com/redhat-developer/vscode-openshift-tools'>
```
Comment (mohitsuman) [inline src/webview/feedback/app/feedback.tsx:53]: Use this URL https://github.com/redhat-developer/vscode-openshift-tools/issues

File: src/webview/feedback/json.js [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+    'pages': [
+        {
+            'name': 'Feedback Page',
+            'elements': [
+                {
+                    'type': 'rating',
+                    'name': 'Satisfaction',
+                    'isRequried': true,
+                    'title': 'Overall, how satisfied or dissatisfied are you with the extension?',
+                    'rateDisplayMode': 'smileys',
```
Comment (mohitsuman) [inline src/webview/feedback/json.js:15]: Let's use numbers 1-5 to rate. As the smiley with heart eyes can only be used for 2-10 rating. For 1-5 rating, it does not represent the correct smileys as expected.

File: src/webview/feedback/json.js [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+                    'titleLocation': 'hidden',
+                    'hideNumber': true,
+                    'placeholder': 'What\'s the main reason for your score?',
+                    'maxLength': 500
+                },
+                {
+                    'type': 'rating',
+                    'name': 'Recommendation',
+                    'title': 'How likely is it that you would recommend the OpenShift Toolkit extension  to a friend or colleague?',
+                    'rateDisplayMode': 'smileys',
```
Comment (mohitsuman) [inline src/webview/feedback/json.js:33]: Similar change with rating based on numbers 1-5.

File: src/webview/feedback/json.js [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+                    'name': 'Used-similar-extension',
+                    'title': 'Have you used similar extension before?'
+                },
+                {
+                    'type': 'text',
+                    'name': 'Similar extension',
+                    'visibleIf': '{Used-similar-extension} = true',
+                    'titleLocation': 'hidden',
+                    'hideNumber': true,
+                    'placeholder': 'Please specify the similar extension...'
```
Comment (mohitsuman) [inline src/webview/feedback/json.js:58]:
```suggestion
                    'placeholder': 'Please mention the similar extension name/URL.'
```

File: src/webview/feedback/json.js [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+                    'visibleIf': '{Recommendation} <= 3',
+                    'titleLocation': 'hidden',
+                    'hideNumber': true,
+                    'placeholder': 'What\'s the main reason for your score?',
+                    'maxLength': 500
+                },
+                {
+                    'type': 'boolean',
+                    'name': 'Used-similar-extension',
+                    'title': 'Have you used similar extension before?'
```
Comment (mohitsuman) [inline src/webview/feedback/json.js:50]:
```suggestion
                    'title': 'Please mention any other similar extension you use together with OpenShift Toolkit?'
```

File: src/webview/feedback/json.js [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+                    'name': 'Similar extension',
+                    'visibleIf': '{Used-similar-extension} = true',
+                    'titleLocation': 'hidden',
+                    'hideNumber': true,
+                    'placeholder': 'Please specify the similar extension...'
+                },
+                {
+                    'type': 'text',
+                    'name': 'Frustrating feature',
+                    'title': 'What, if anything, do you find frustrating or unappealing about the extension ?'
```
Comment (mohitsuman) [inline src/webview/feedback/json.js:63]:
```suggestion
                    'title': 'What, if anything, do you find frustrating or challenging about the extension workflow?
```

File: src/webview/feedback/json.js [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+                    'type': 'text',
+                    'name': 'Best feature',
+                    'title': 'What do you like best about the extension?'
+                },
+                {
+                    'type': 'text',
+                    'name': 'contact',
+                    'hideNumber': true,
+                    'title': 'Share your contact information if you\'d like us to answer you:',
+                    'placeholder': 'Enter email address, phone number or leave blank for anonymous feedback'
```
Comment (mohitsuman) [inline src/webview/feedback/json.js:80]:
```suggestion
                    'placeholder': 'Provide email address or leave blank for anonymous feedback'
```

**Review by mohitsuman — CHANGES_REQUESTED** [review]
Some wording changes

File: src/webview/feedback/app/feedback.tsx [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+                                </Stack>
+                            </div>
+                        </a>
+                        <a href='https://marketplace.visualstudio.com/items?itemName=redhat.vscode-openshift-connector&ssr=false#review-details'>
+                            <div className='section__header-hint section__footer'>
+                                <Stack direction='row' alignItems='center' gap={1}>
+                                    <Icon fontSize='small'>
+                                        <img src={MicrosoftIcon} />
+                                    </Icon>
+                                    <Typography variant='body2' className='footerText'>Rate us on Marketplace</Typography>
```
Comment (mohitsuman) [inline src/webview/feedback/app/feedback.tsx:71]: The animation of underlining does not happen in the text. It happens on the MSFT logo. it works well for the GitHub icon and text.

File: src/webview/feedback/app/feedback.tsx [hunk 12]
```diff
@@ -0,0 +1,85 @@
...
+                                </Stack>
+                            </div>
+                        </a>
+                    </label>
+                </div>
+            </div>
+            <div className='captionHeader'>
+                <Typography variant='subtitle2' className='foreGroundColor'>The Red Hat OpenShift Toolkit extension team would like to learn from your experience to improve the extension workflow.</Typography>
+                <Typography variant='subtitle2' className='foreGroundColor'>This survey will take about 3 minutes. Your feedback is extremely valuable and will directly impact the product moving forward.</Typography>
```
Comment (mohitsuman) [inline src/webview/feedback/app/feedback.tsx:80]:
```suggestion
                <Typography variant='subtitle2' className='foreGroundColor'>This survey will take about 2 minutes. Your feedback is extremely valuable and will directly impact the product moving forward.</Typography>
```

File: src/webview/feedback/app/index.html [hunk 13]
```diff
@@ -0,0 +1,52 @@
+<!DOCTYPE html>
+<html lang="en">
+    <head>
+        <base href="%BASE_URL%" />
+        <meta charset="UTF-8" />
+        <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+        <title>Feedback</title>
```
Comment (mohitsuman) [inline src/webview/feedback/app/index.html:7]:
```suggestion
        <title>Share Feedback</title>
```

**Review by mohitsuman — CHANGES_REQUESTED** [review]
Few changes.

File: src/webview/feedback/app/index.html (hunk 13 above)
Comment (mohitsuman) [reply src/webview/feedback/app/index.html]: @msivasubramaniaan is this done ?

Comment (msivasubramaniaan) [reply src/webview/feedback/app/index.html]: yes

**Review by mohitsuman — APPROVED** [review]
LGTM

**Review by datho7561 — APPROVED** [review]
Looks good! Thanks Muthu!

**Pull request merged**